crates/common/tedge_config/src/tedge_config_cli/new.rs

```diff
@@ -433,7 +433,6 @@ define_tedge_config! {
             #[doku(as = "u16")]
             port: NonZeroU16,
 
-            #[tedge_config(reader(private))]
```

Comment:
To access the `auth.cert and key files` had to remove this private.

Reply:
Before removing this, I would ask @jarhodes314 about why he added this (taking the time to implement this macro only for this `auth` field and the c8y url). 

If this reader has to be kept private, a solution is to add methods in `new::TEdgeConfig` as this has been done to [get the `c8y_url`](https://github.com/thin-edge/thin-edge.io/blob/main/crates/common/tedge_config/src/tedge_config_cli/new.rs#L54)

Reply:
The goal here was to stop people accessing these fields directly because `auth.key_file` and `auth.cert_file` must either both be set, or neither of them should be set. The idea here is that whenever we access the two fields, we need to return an error if only one of them is set. If you look at https://github.com/thin-edge/thin-edge.io/blob/b044aaebf0f643e353ce32e62518236e5de09b25/crates/common/tedge_config/src/tedge_config_cli/new.rs#L77-L80, for example, you'll see we use `all_or_nothing` to access the fields. I think the appropriate solution in this case is to create a method on TedgeConfigMqttReader (or whatever is called, I'd look at the `cargo doc` output to figure that out) called something like `cert_key_files()` that returns the result of calling all_or_nothing.

Once you've done that, you can probably modify https://github.com/thin-edge/thin-edge.io/pull/2077/files#diff-e932087e82e2186b8e86bcbdc49124d1e279770b5f055c33d69f832ff4f1101dR50-R55 to use `if let Some((cert_file, key_file)) = config.mqtt.cert_key_files() { ... }` rather than needing to unwrap in the where the cert and key and some.

Reply:
Thanks for the clarification @jarhodes314

Reply:
The updated code looks much better.